### DISCUSSION TITLE
Is it possible to reload Gradle dependencies via nvim-jdtls?

### DISCUSSION THREAD

**Discussion by TheBlob42** [mfussenegger/nvim-jdtls#127] [General]
Eclipse has the "Refresh Gradle Project" option, which checks the projects Gradle configuration and updates the dependencies and tasks accordingly (e.g. updating to newer version, loaded newly added dependencies etc.) 

Currently I need to delete the `.project` file + delete the project from workspace and then restart JDTLS in order to consistently trigger such an "update". And even this does not work without issues from time to time.

Therefore I thought: Maybe there is a way to trigger this action on JDTLS via `nvim-jdtls` already. Is it even possible using the eclipse LSP :slightly_smiling_face:

**Comment by mfussenegger** [comment]
`:JdtUpdateConfig` (or `require('jdtls').update_project_config()` if you didn't call `add_commands`) should re-read the configuration. But I noticed that depending on the changes made to the gradle configuration I noticed that eclipse.jdt.ls sometimes isn't able to apply all changes and you've to go through the restart procedure anyways.

**Reply by TheBlob42** [reply]
I forgot to mention that I have tried `JdtUpdateConfig` already. But when adding a new dependency to my `build.gradle` or updating the version of a dependency this is unfortunately not enough and I __always__ need to restart eclipse.jdt.ls in order to fetch the new dependencies. 

That's why I thought: maybe there is a more specific command like 'Refresh Gradle Project' which works better in this regard :thinking: 

Maybe I need to check what you're sending to the LSP on `JdtUpdateConfig` and ask in the eclipse.jdt.ls repository why this sometimes does not work or if there is a "better way" to trigger such an update for Gradle.

**Reply by clement-buchart** [reply]
Hello @TheBlob42 
If your project has subprojects somewhat like described [here](https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:adding_subprojects), be aware that calling `JdtUpdateConfig` on a file in a subproject will only refresh the `build.gradle` for this particular subproject.

**Reply by Andrew15-5** [reply]
@clement-buchart, if I choose `gradle init`, `application` then it will add a single subproject `app`. There is only one `build.gradle` in the whole project. And the dependency only goes in that `build.gradle`. I also (as OP) can really update config only after executing `:JdtWipeDataAndRestart` — `:JdtUpdateConfig` and `:JdtRestart` (or `:LspRestart`) fix nothing.

**Comment by EnriqueBravo115** [comment]
Hey bro you found a well solution for this problem?

**Reply by TheBlob42** [reply]
What always works for me is to use the `JdtWipeDataAndRestart` command.
This way I don't have to fiddle with the .project files and a manual restart myself. Since this is working fine nowadays I haven't invested more time into looking this :slightly_smiling_face:

**Reply by EnriqueBravo115** [reply]
Bro i love you now i can delete intellij(not really)

**Reply by anafro** [reply]
> What always works for me is to use the `JdtWipeDataAndRestart` command. This way I don't have to fiddle with the .project files and a manual restart myself. Since this is working fine nowadays I haven't invested more time into looking this 🙂

Let God give you everything you wish, you solved the biggest issue I had!!!
